intellij-plugin-structure/structure-ide/src/main/java/com/jetbrains/plugin/structure/ide/dependencies/PluginXmlDependencyFilter.kt

```diff
@@ -0,0 +1,52 @@
+package com.jetbrains.plugin.structure.ide.dependencies
+
+import com.jetbrains.plugin.structure.base.utils.closeAll
+import com.jetbrains.plugin.structure.xml.ElementNamesFilter
+import com.jetbrains.plugin.structure.xml.EventTypeExcludingEventFilter
+import com.jetbrains.plugin.structure.xml.LogicalAndXmlEventFilter
+import com.jetbrains.plugin.structure.xml.newEventWriter
+import com.jetbrains.plugin.structure.xml.newFilteredEventReader
+import java.io.Closeable
+import java.io.IOException
+import java.io.InputStream
+import java.io.OutputStream
+import javax.xml.stream.EventFilter
+import javax.xml.stream.XMLInputFactory
+import javax.xml.stream.XMLOutputFactory
+import javax.xml.stream.events.XMLEvent
+import javax.xml.stream.events.XMLEvent.COMMENT
+import javax.xml.stream.events.XMLEvent.START_DOCUMENT
+
+class PluginXmlDependencyFilter(private val ignoreComments: Boolean = true, private val ignoreXmlDeclaration: Boolean = true) {
```

Comment:
This is to take an XML and retain only relevant elements. Most importantly `<depends>` and `<dependencies>`. This leaves us with an IDE that has 0 classes and just a bare-bones plugin descriptor structure that indicates a dependency relationships between plugins and modules.

This is just for tests, to immitate a real-life IDE.

I consider the type-safe IDE model in the `structure` to be not-quite-bug-free, especially with v2 model. Dumping XML from the model might lose some information.

Reply:
This can be merged with a plan to migrate away from JAXB to Kotlin Serialization. (This is blocked by technical issues on the Kotlin 1.4 side).

I will leave this now, and consolidate efforts with that initiative.